app/views/users/_form.html.erb

```diff
@@ -0,0 +1,15 @@
+<%= f.input :email %>
+
+<% if @user.provider %>
+  <div class="alert alert-warning">
+    This person signs in with <%= @user.provider %>
+  </div>
+<% else %>
+  <%= f.input :password %>
```

Comment:
@braddoeswebdev should we do "old password" before they reset their password with "new password" and "confirm new password"? if someone else has access to some user's device it would be extremely easy to change it just like that..

Reply:
Not a bad idea.